src/scoring/hpf_scoring.py

```diff
@@ -0,0 +1,168 @@
+"""The HPF Model scoring class."""
+
+import numpy as np
+import os
+from edward.models import Poisson
+from edward.models import Gamma
+import tensorflow as tf
+from collections import defaultdict
+from src.config import (SCORING_THRESHOLD,
+                        HPF_SCORING_REGION,
+                        HPF_output_package_id_dict,
+                        HPF_output_manifest_id_dict,
+                        HPF_output_rating_matrix,
+                        HPF_output_item_matrix,
+                        a, a_c, c, c_c,
+                        b_c, d_c, K)
+
+
+class HPFScoring:
+    """The HPF Model scoring class."""
+
+    def __init__(self, datastore=None, scoring_threshold=SCORING_THRESHOLD,
+                 scoring_region=HPF_SCORING_REGION):
+        """Set the variables and load model data."""
+        self.datastore = datastore
+        self.scoring_threshold = scoring_threshold
+        self.scoring_region = scoring_region
+        self.package_id_dict = None
+        self.id_package_dict = None
+        self.rating_matrix = None
+        self.beta = None
+        self.manifest_id_dict = None
+        self.manifests = 0
+        self.packages = 0
+        self.rating_mean = 0.0
+        self.a = tf.constant(a)
+        self.a_c = tf.constant(a_c)
+        self.c = tf.constant(c)
+        self.c_c = tf.constant(c_c)
+        self.b_c = tf.constant(b_c)
+        self.d_c = tf.constant(d_c)
+        self.K = K
+        self.loadS3()
+
+    def loadS3(self):
+        """Load the model data from AWS S3."""
+        package_id_dict_filename = os.path.join(
+            self.scoring_region, HPF_output_package_id_dict)
+        self.package_id_dict = self.datastore.read_json_file(
+            package_id_dict_filename)
+        self.id_package_dict = {x: n for n, x in self.package_id_dict.items()}
+        manifest_id_dict_filename = os.path.join(
+            self.scoring_region, HPF_output_manifest_id_dict)
+        self.manifest_id_dict = self.datastore.read_json_file(
+            manifest_id_dict_filename)
+        self.manifest_id_dict = {n: set(x)
+                                 for n, x in self.manifest_id_dict.items()}
+        rating_matrix_filename = os.path.join(
+            self.scoring_region, HPF_output_rating_matrix)
+        self.datastore.download_file(
+            rating_matrix_filename, "/tmp/rating_matrix.npy")
+        self.rating_matrix = np.load("/tmp/rating_matrix.npy")
```

Comment:
I would suggest storing all of these as compressed `.npz` files instead of `.npy` since our matrix is very sparse a 8 GB npy file is around 1.2 MB when compressed.